internal/worker/clienterrors/errors.go

```diff
@@ -0,0 +1,84 @@
+package clienterrors
+
+import (
+	"fmt"
+)
+
+const (
+	ErrorCodePrefix = "IMAGE-BUILDER-WORKER-CLIENT-"
+
+	ErrorNoDynamicArgs       ClientErrorCode = 9
+	ErrorInvalidTargetConfig ClientErrorCode = 10
+	ErrorSharingTarget       ClientErrorCode = 11
+	ErrorInvalidTarget       ClientErrorCode = 12
+
+	ErrorDNFError             ClientErrorCode = 1001
+	ErrorDepsolveJob          ClientErrorCode = 1002
+	ErrorDepsolveDependency   ClientErrorCode = 1003
+	ErrorParsingDynamicArgs   ClientErrorCode = 1004
+	ErrorManifestGeneration   ClientErrorCode = 1005
+	ErrorManifestDependency   ClientErrorCode = 1006
+	ErrorBuildJob             ClientErrorCode = 1007
+	ErrorUploadingImage       ClientErrorCode = 1008
+	ErrorRegisteringImage     ClientErrorCode = 1009
+	ErrorKojiFailedDependency ClientErrorCode = 10010
+	ErrorKojiBuild            ClientErrorCode = 1011
+	ErrorKojiInit             ClientErrorCode = 1012
+	ErrorKojiFinalize         ClientErrorCode = 1013
+	ErrorInvalidConfig        ClientErrorCode = 1014
+
+	// Errors contained within this file
+	ErrorClientErrorNotFound ClientErrorCode = 10002
+)
+
+type ClientErrorCode int
+
+type ClientError struct {
+	Code   ClientErrorCode
+	Reason string
+}
+
+var ClientErrors = map[ClientErrorCode]ClientError{
+	ErrorNoDynamicArgs:        {ErrorNoDynamicArgs, "No dynamic arguments"},
+	ErrorInvalidTargetConfig:  {ErrorInvalidTargetConfig, "Provided user config is invalid"},
+	ErrorSharingTarget:        {ErrorSharingTarget, "Error sharing image with provided account"},
+	ErrorInvalidTarget:        {ErrorInvalidTarget, "Unsupported target"},
+	ErrorDNFError:             {ErrorDNFError, "A DNF error occurred"},
+	ErrorDepsolveJob:          {ErrorDepsolveJob, "Error completing depsolve job"},
+	ErrorDepsolveDependency:   {ErrorDepsolveDependency, "Error in depsolve job dependency"},
+	ErrorParsingDynamicArgs:   {ErrorParsingDynamicArgs, "Unable to parse dynamic args"},
+	ErrorManifestGeneration:   {ErrorManifestGeneration, "Error generating image manifest"},
+	ErrorBuildJob:             {ErrorBuildJob, "Error with osbuild job"},
+	ErrorUploadingImage:       {ErrorUploadingImage, "Error during image upload to target"},
+	ErrorRegisteringImage:     {ErrorRegisteringImage, "Error registering image to target"},
+	ErrorKojiFailedDependency: {ErrorKojiFailedDependency, "Koji job has a failed dependency"},
+	ErrorKojiBuild:            {ErrorKojiBuild, "Koji build error"},
+	ErrorKojiInit:             {ErrorKojiInit, "Koji init error"},
+	ErrorKojiFinalize:         {ErrorKojiFinalize, "Koji finalize error"},
+	ErrorInvalidConfig:        {ErrorInvalidConfig, "Configuration error"},
+	ErrorClientErrorNotFound:  {ErrorClientErrorNotFound, "Error does not exist"},
+}
+
+type Error struct {
+	Kind    string          `json:"kind"`
+	ID      ClientErrorCode `json:"id"`
+	Code    string          `json:"code"`
```

Comment:
Same with code, I think in this case just the `id` is enough.

Reply:
I agree with this, it was starting to seem very redundant.